Client/Frontend/Menu/Items/MenuItemView.swift

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+/**
+ * creates a view that consists of an image and a title.
+ * the image view is displayed at the center top of the
+ **/
+public class MenuItemView: UIControl {
+
+    var padding: CGFloat = 5.0
+
+    private(set) public var imageView: UIImageView
+    private(set) public var titleLabel: UILabel
+
+    private var previousLocation: CGPoint?
+    public init() {
+        imageView = UIImageView()
+        titleLabel = UILabel()
+
+        super.init(frame: CGRectZero)
+
+        titleLabel.lineBreakMode = NSLineBreakMode.ByWordWrapping
+        titleLabel.numberOfLines = 2
+        titleLabel.textAlignment = NSTextAlignment.Center
+
+        self.addSubview(imageView)
+
+        self.addSubview(titleLabel)
+
+        imageView.snp_makeConstraints { make in
+            make.centerX.equalTo(self)
+            make.bottom.equalTo(self.snp_centerYWithinMargins)
+        }
+
+        titleLabel.snp_makeConstraints { make in
+            make.top.equalTo(imageView.snp_bottom).offset(padding)
+            make.centerX.equalTo(self)
+            make.bottom.equalTo(self).offset(-padding)
+            make.width.lessThanOrEqualTo(self.snp_width).dividedBy(1.5)
+        }
```

Comment:
So, I updated the constraints here to:
```
        customImageView.snp_makeConstraints { make in
            make.centerX.equalTo(self)
            make.top.equalTo(self).offset(2 * padding)
        }

        customTitleLabel.snp_makeConstraints { make in
            make.top.greaterThanOrEqualTo(customImageView.snp_bottom).offset(padding/2)
            make.centerX.equalTo(self)
            make.left.equalTo(self).offset(padding)
            make.right.equalTo(self).offset(-padding)
            make.bottom.greaterThanOrEqualTo(self).offset(-padding)
        }
```

The result of which was a good alignment, however, the single line rows need to be aligned centrally between the image and the bottom of the view, so it appears in the center of the 2 line text rather than at the bottom. Not really sure how to achieve that here (that was kind what all the fancy division was all about). Any ideas?

How it was:
![screen shot 2016-03-21 at 4 46 31 pm](https://cloud.githubusercontent.com/assets/12613048/13926507/8648b3f4-ef84-11e5-93ff-8256e3c1e6f5.png)

Now:

![screen shot 2016-03-21 at 4 44 39 pm](https://cloud.githubusercontent.com/assets/12613048/13926441/4245bb70-ef84-11e5-9666-f1f643383bc6.png)